src/lib.rs

```diff
@@ -12,99 +12,155 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-extern crate termios;
-extern crate libc;
-
-use libc::consts::os::posix88::STDIN_FILENO;
-use std::old_io::stdio::{ stdin, StdinReader };
-use std::old_io::BufReader;
-use std::old_io::IoResult;
-use std::slice::bytes::MutableByteVector;
-
-#[cfg(test)]
-static TEST_BUFFER: &'static [u8] = b"my-secret\npassword";
-
-#[cfg(test)]
-static mut TEST_EOF: bool = false;
-
-#[cfg(test)]
-static mut TEST_HAS_SEEN_EOF_BUFFER: bool = false;
-
-#[cfg(test)]
-static mut TEST_HAS_SEEN_REGULAR_BUFFER: bool = false;
-
-
-#[cfg(test)]
-fn get_reader() -> BufReader<'static> {
-    if unsafe { TEST_EOF } {
-        unsafe { TEST_HAS_SEEN_EOF_BUFFER = true; }
-        let mut reader = BufReader::new(b"");
-        reader.read_to_end().unwrap();
-        reader
-    } else {
-        unsafe { TEST_HAS_SEEN_REGULAR_BUFFER = true; }
-        BufReader::new(TEST_BUFFER)
+#![cfg_attr(windows, feature(io))]
+
+#[cfg(not(windows))]
+mod unix {
+    extern crate termios;
+    extern crate libc;
+
+    use libc::consts::os::posix88::STDIN_FILENO;
```

Comment:
For some reason, this now needs to be `use libc::consts::os::posix88::STDIN_FILENO` I think. It doesn't build on UNIX in the current state.